File: documentation/manual/working/javaGuide/main/tests/code/javaguide/tests/controllers/HomeController.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package javaguide.tests.controllers;

import play.mvc.*;

public class HomeController extends Controller {

  public Result index() {
    return ok(javaguide.tests.html.index.render("Welcome to Play!"));
  }

  public Result post(Http.Request request) {
    return redirect(routes.HomeController.index());
  }
}

File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/evolution-path-config/app/controllers/HomeController.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package controllers

import javax.inject._

import play.api._
import play.api.db.Database
import play.api.mvc._

@Singleton
class HomeController @Inject() (val controllerComponents: ControllerComponents, val database: Database)
    extends BaseController {

  def index() = Action { implicit request: Request[AnyContent] =>
    val value = database.withConnection(connection => {
      val rs = connection
        .prepareStatement("select * from table1")
        .executeQuery()
      rs.next()
      rs.getString(1)
    })
    Ok(value)
  }
}
